### PR TITLE
Add vararg support

### PR DESCRIPTION
**Pull request by dykov**
Add varargs support for '.all', '.any', '.none', '.select' methods

-----
[View rendered readme.md](https://github.com/dykov/enumerable4j/blob/vararg-support/readme.md)

### CODE REVIEW

File: src/main/java/io/github/dgroup/enumerable4j/Enumerable.java [hunk 1]
```diff
@@ -62,6 +62,20 @@ default boolean all(Predicate<X> prd) {
         return match;
     }
 
+    /**
+     * Passes each element of the collection to the each given blocks.
+     * @param first The predicate to match each element.
+     * @param other The array of predicates to match each element.
+     * @return The true if the blocks never return false or nil.
+     */
+    default boolean all(Predicate<X> first, Predicate<X>... other) {
```
Comment (dgroup) [inline src/main/java/io/github/dgroup/enumerable4j/Enumerable.java:71]: no need to add new methods, you should change old methods.
Please add everywhere `Predicate<X> ... other`

**Review by dgroup — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/dgroup/enumerable4j/Enumerable.java (hunk 1 above)
Comment (dykov) [reply src/main/java/io/github/dgroup/enumerable4j/Enumerable.java]: done.

File: readme.md [hunk 2]
```diff
@@ -80,6 +90,17 @@ public interface Enumerable<X> extends Collection<X> {
...
     }
 
+    /**
+     * Passes at least one element of the collection to the each given blocks.
+     * If no predicate (null) is given, then true is returned instead.
+     * @param first The predicate to match at least one element.
+     * @param other The array of predicates to match at least one element.
+     * @return The true if the blocks never return false or nil.
+     */
+    default boolean any(Predicate<X> first, Predicate<X>... other) {
```
Comment (dgroup) [inline readme.md:100]: please remove the obsolete methods from `readme.md` also

**Review by dgroup — CHANGES_REQUESTED** [review]

File: readme.md (hunk 2 above)
Comment (dykov) [reply readme.md]: done

**Review by dgroup — CHANGES_REQUESTED** [review]
varargs allows to support 0-length parameters like select() without predicate.

please keep select(first, ... other) schema

File: src/main/java/io/github/dgroup/enumerable4j/Enumerable.java [hunk 3]
```diff
@@ -47,69 +47,89 @@
...
-     * Passes at least one element of the collection to the given block.
+     * Passes at least one element of the collection to the each given function.
      * If no predicate (null) is given, then true is returned instead.
-     * @param prd The predicate to match at least one element.
-     * @return The true if the block never returns false or nil.
+     * @param prd The array of functions to match at least one element.
+     * @return True if the functions never return false or nil.
      */
-    default boolean any(Predicate<X> prd) {
+    default boolean any(Predicate<X>... prd) {
```
Comment (dgroup) [inline src/main/java/io/github/dgroup/enumerable4j/Enumerable.java:73]: please use `Predicate<T> first, Predicate<T> ...other` everywhere.

Comment (dykov) [reply src/main/java/io/github/dgroup/enumerable4j/Enumerable.java:73]: what should be if any of the predicates is null?
should it throw any exception? or should it return true?

Comment (dykov) [reply src/main/java/io/github/dgroup/enumerable4j/Enumerable.java:73]: for example, what should the next method return?
```java
src.any( pred , null, null, null);
```

Comment (dgroup) [reply src/main/java/io/github/dgroup/enumerable4j/Enumerable.java:73]: i think let's just skip the `null` based predicates

Comment (dykov) [reply src/main/java/io/github/dgroup/enumerable4j/Enumerable.java:73]: So, if all arguments are ‘null’ the result is ‘true’. right?

Comment (dykov) [reply src/main/java/io/github/dgroup/enumerable4j/Enumerable.java]: @dgroup , please check the new implementation. 
If it’s ok merge the changes please. 
The implementation of `find`, `next` etc methods should be discussed later

**Review by dgroup — APPROVED** [review]

**Comment by dgroup** [comment]
@rultor merge

**Pull request merged**